### ISSUE TITLE
UgilfyJS - Unexpected token: name (Simple) 

### ISSUE THREAD

**Issue by adrlen** [hildjj/node-cbor#57]
This library is used by [autobahn-js](https://github.com/crossbario/autobahn-js).
When adding autobahn to my vue-js app, I have an error when trying to build for production using webpack 2.6 and UglifyJS :

`ERROR in static/js/app.bdcd88ac9fe0000fe77f.js from UglifyJs`
`Unexpected token: name (Simple) [./~/cbor/lib/simple.js:9,0]`

**Comment by mtancoigne** [comment]
I first bypassed this adding cbor in the babel-loader include section of webpack config, but it appears that the build is broken in the end...

**Comment by hildjj** [comment]
As far as I can tell from https://github.com/crossbario/autobahn-js/issues/306, this is because your system doesn't support ES6 classes. If that's the case, whatever system that is, I'm not interested in supporting it directly.  You might be able to use babel to transpile to an ancient javascript version if that's what you want.  I'll leave this open for a bit in case you want to talk me into a particular fix.

**Comment by hildjj** [comment]
Having heard nothing further, WONTFIX.  Feel free to comment more and I'll reopen if needed.

**Comment by klappy** [comment]
The problem I am encountering is that it can't successfully transpile using babel and webpack. We ran into this issue with some of our own NPM packages. 

There are lots of best practice blog posts and google searches that led us to believe that it is standard best practice to transpile before publishing to help avoid issues in libraries that depend on them.  Don't take my word for it, I just did some quick google searching and there are many results for this.

- https://www.wouterbulten.nl/blog/tech/publishing-es6-package-to-npm/
> For private projects the transpiler can be incorporated in to the build process. But, how do we proceed when we want to make a project public and usable by others? This is particular important when we publish a package to NPM. NPM packages should run on “ES5 systems”.

- https://medium.com/@tarkus/how-to-build-and-publish-es6-modules-today-with-babel-and-rollup-4426d9c7ca71
- https://hackernoon.com/how-to-publish-your-package-on-npm-7fc1f5aae600
